### PR TITLE
Use path relative to the directory of Reanimated when checking its version.

### PR DESCRIPTION
**Pull request by j-piasecki**
## Description

Use path relative to the directory of Reanimated when checking its version.

### CODE REVIEW

**Review by kmagiera — APPROVED** [review]

**Comment by HeroBanana** [comment]
> ## Description
> Use path relative to the directory of Reanimated when checking its version.

Will do

**Comment by Hirbod** [comment]
Can you guys make a release for this please?

**Comment by sorodrigo** [comment]
Could we make a new release with this change? Please, pretty please 🙏🏻 🙇🏻‍♂️ .

**Comment by HeroBanana** [comment]
You guys can use a patch with https://github.com/ds300/patch-package. 

Example of my patch: react-native-gesture-handler+2.1.0.patch
```
diff --git a/node_modules/react-native-gesture-handler/android/build.gradle b/node_modules/react-native-gesture-handler/android/build.gradle
index 8afc3d5..15a6a70 100644
--- a/node_modules/react-native-gesture-handler/android/build.gradle
+++ b/node_modules/react-native-gesture-handler/android/build.gradle
@@ -21,8 +21,9 @@ def safeExtGet(prop, fallback) {
 
 // Check whether Reanimated 2.3 or higher is installed alongside Gesture Handler
 def shouldUseCommonInterfaceFromReanimated() {
-    if (rootProject.subprojects.find { it.name == 'react-native-reanimated' } != null) {
-        def inputFile = new File(rootProject.projectDir, '../node_modules/react-native-reanimated/package.json')
+    def reanimated = rootProject.subprojects.find { it.name == 'react-native-reanimated' };
+    if (reanimated != null) {
+        def inputFile = new File(reanimated.projectDir, '../package.json')
         def json = new JsonSlurper().parseText(inputFile.text)
         def reanimatedVersion = json.version as String
         def (major, minor, patch) = reanimatedVersion.tokenize('.')

```

**Comment by SrBrahma** [comment]
Please, release it!

**Comment by SrBrahma** [comment]
I've tried to `yarn add react-native-gesture-handler@https://github.com/software-mansion/react-native-gesture-handler` as this is already inculded in main,

but I've got this error:

```
> Task :app:compileDebugJavaWithJavac
[RUN_GRADLEW] /tmp/hb/eas-build-local-nodejs/474f5a6c-3e09-4724-8fd6-9b022984562d/build/app/android/app/build/generated/rncli/src/main/java/com/facebook/react/PackageList.java:29: error: cannot find symbol
[RUN_GRADLEW] import com.swmansion.gesturehandler.RNGestureHandlerPackage;
[RUN_GRADLEW]                                    ^
[RUN_GRADLEW]   symbol:   class RNGestureHandlerPackage
[RUN_GRADLEW]   location: package com.swmansion.gesturehandler
[RUN_GRADLEW] /tmp/hb/eas-build-local-nodejs/474f5a6c-3e09-4724-8fd6-9b022984562d/build/app/android/app/build/generated/rncli/src/main/java/com/facebook/react/PackageList.java:94: error: cannot find symbol
[RUN_GRADLEW]       new RNGestureHandlerPackage(),
[RUN_GRADLEW] ^
[RUN_GRADLEW]   symbol:   class RNGestureHandlerPackage
[RUN_GRADLEW]   location: class com.facebook.react.PackageList
[RUN_GRADLEW] Note: /tmp/hb/eas-build-local-nodejs/474f5a6c-3e09-4724-8fd6-9b022984562d/build/app/android/app/src/debug/java/com/tekhni/ahive/ReactNativeFlipper.java uses or overrides a deprecated API.
[RUN_GRADLEW] Note: Recompile with -Xlint:deprecation for details.
[RUN_GRADLEW] 2 errors
[RUN_GRADLEW] > Task :app:compileDebugJavaWithJavac FAILED
[RUN_GRADLEW] w: Detected multiple Kotlin daemon sessions at build/kotlin/sessions
[RUN_GRADLEW] FAILURE: Build failed with an exception.
[RUN_GRADLEW] * What went wrong:
[RUN_GRADLEW] Execution failed for task ':app:compileDebugJavaWithJavac'.
[RUN_GRADLEW] > Compilation failed; see the compiler error output for details.
[RUN_GRADLEW] * Try:
[RUN_GRADLEW] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[RUN_GRADLEW] * Get more help at https://help.gradle.org
[RUN_GRADLEW] BUILD FAILED in 11m 39s
[RUN_GRADLEW] Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
[RUN_GRADLEW] Use '--warning-mode all' to show the individual deprecation warnings.
[RUN_GRADLEW] See https://docs.gradle.org/6.9/userguide/command_line_interface.html#sec:command_line_warnings
[RUN_GRADLEW] 504 actionable tasks: 504 executed
[RUN_GRADLEW] Error: Gradle build failed with unknown error. Please see logs for the "Run gradlew" phase.

Build failed
Gradle build failed with unknown error. Please see logs for the "Run gradlew" phase.
    Error: npx exited with non-zero code: 1
```

**Comment by sorodrigo** [comment]
Can confirm the patch works. 🏆

**Comment by veeramarni** [comment]
What version have this fix?

**Comment by SrBrahma** [comment]
Not released yet :/

**Comment by Knaackee** [comment]
Any news on release date?

**Comment by Knaackee** [comment]
I tried the patch, but it didn't work for me (build on EAS) so I tried to find the problem.

## Problem

I got an error on EAS build.

```
// ...
> Configure project :expo
Using expo modules
- expo-application (4.0.1)
  - expo-barcode-scanner (11.2.0)
  - expo-camera (12.1.0)
  - expo-constants (13.0.0)
  - expo-dev-client (0.8.0)
  - expo-dev-launcher (0.10.0)
  - expo-dev-menu (0.9.0)
  - expo-file-system (13.1.0)
  - expo-font (10.0.4)
  - expo-image-loader (3.1.0)
  - expo-json-utils (0.2.0)
  - expo-keep-awake (10.0.1)
  - expo-manifests (0.2.2)
  - expo-modules-core (0.4.10)
  - expo-splash-screen (0.14.1)
  - react-native-reanimated (2.3.1)
[stderr] FAILURE: Build failed with an exception.
[stderr] * Where:
[stderr] Build file '/root/workingdir/build/node_modules/react-native-gesture-handler/android/build.gradle' line: 26
[stderr] * What went wrong:
[stderr] A problem occurred evaluating project ':react-native-gesture-handler'.
[stderr] > /root/workingdir/build/packages/@myorg/mobile/android/../node_modules/react-native-reanimated/package.json (No such file or directory)
[stderr] * Try:
[stderr] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 1m 3s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.9/userguide/command_line_interface.html#sec:command_line_warnings
Error: Gradle build failed with unknown error. Please see logs for the "Run gradlew" phase.
```


What I did:

1. Tried the [patch](https://github.com/software-mansion/react-native-gesture-handler/pull/1757#issuecomment-998573245).
2.  Still got the error. So, I added the following to the `postinstall` script `patch-package && echo \"$(cat ./node_modules/react-native-gesture-handler/android/build.gradle)\` to check if my patch get applied on EAS. It did. Still the same error. Still on `line: 26`.
3. I changed the `build.gradle` (added comments, even removed `shouldUseCommonInterfaceFromReanimated` and `dependencies` entry. I then created a new patch. Still the same. My output is correct, the file gets changed as expected but i still get the error Still on `line: 26`.

So, in short, it looks that for some reason my `build.gradle` doesn't get used. I checked all the paths. I have only one `react-native-gesture-handler` package installed.

I am currently stuck, any ideas?

**Comment by SrBrahma** [comment]
@Knaackee try to use `yarn patch`. I had an error with patch-package with yarn, as it won't work with yarn > 2. Maybe it was the same error as yours.

I also use expo eas and got it working with yarn patch. I don't remember exactly how I did it but got it working quickly.

**Comment by Knaackee** [comment]
@SrBrahma thanks for your reply! EAS uses `Yarn 1.22.10` so this could not be the problem. 

Very strange

**Comment by Knaackee** [comment]
@SrBrahma thanks for your reply! 

I am a little confused. 

Locally, I am using yarn `1.22.17` and eas uses `1.22.10` and not v2. 
`patch-package` works fine with yarn v 1.x. `yarn patch` is a yarn v2 feature.



Very strange

**Comment by SrBrahma** [comment]
I see. So unfortunately I can not help you :/ Good luck!

**Comment by Knaackee** [comment]
Update:

`eas build --profile development --platform android`, which tries to produce an `Android internal distribution build`, fails but `expo build:android`, which does produce a `Android Play Store build` works. 

Why?

🤔

**Comment by SrBrahma** [comment]
This is now included in 2.2.0!

**Comment by veeramarni** [comment]
thanks @SrBrahma! did a quick test and now the build work fine.
